cosmwasm/enclaves/shared/contract-engine/src/wasm3/mod.rs

```diff
@@ -1755,3 +1757,18 @@ fn get_encryption_salt(timestamp: u64) -> Vec<u8> {
 
     encryption_salt
 }
+
+const GAS_MULTIPLIER: u64 = 1000;
+
+fn host_gas_evaporate(
+    context: &mut Context,
+    instance: &wasm3::Instance<Context>,
+    evaporate: i32,
+) -> WasmEngineResult<i32> {
+    let evaporate = (evaporate as u32) as u64 * GAS_MULTIPLIER;
```

Comment:
Oh, btw this needs a check if evaporate is 0 and if not to evaporate some base gas (specified in gas_costs) - otherwise you can DoS the chain by just calling this endlessly with 0 gas evaporated